File: Javascript/grid.js
class Grid {
    constructor(rows, cols) {
        this.rows = rows;
        this.cols = cols;
        this.currentGrid = Array.from({ length: rows }, () => Array(cols).fill(false));
        this.nextGrid = Array.from({ length: rows }, () => Array(cols).fill(false));
    }

    getRows() {
        return this.rows;
    }

    getCols() {
        return this.cols;
    }

    getCurrentGrid() {
        return this.currentGrid;
    }

    display() {
        for (let r = 0; r < this.rows; r++) {
            let line = "";
            for (let c = 0; c < this.cols; c++) {
                line += this.currentGrid[r][c] ? '*' : '.';
                line += ' ';
            }
            console.log(line);
        }
    }

    update() {
        for (let r = 0; r < this.rows; r++) {
            for (let c = 0; c < this.cols; c++) {
                let aliveNeighbors = this.countAliveNeighbors(r, c);
                if (this.currentGrid[c][r]) {
                    this.nextGrid[r][c] = (aliveNeighbors === 2 || aliveNeighbors === 3);
                } else {
                    this.nextGrid[r][c] = (aliveNeighbors === 3);
                }
            }
        }
        this.currentGrid = this.nextGrid.map(arr => [...arr]);
    }

    setCell(row, col, alive) {
        if (this.rows > 0 && row < this.rows && this.cols > 0 && col < this.cols) {
            this.currentGrid[row][col] = alive;
        }
    }

    countAliveNeighbors(row, col) {
        const dirs = [-1, 0, 1];
        let count = 0;
        for (let dr of dirs) {
            for (let dc of dirs) {
                let r = row + dr;
                let c = col + dc;
                if (r >= 0 && r < this.rows && c >= 0 && c < this.cols && this.currentGrid[r][c]) {
                    count++;
                }
            }
        }
        return count;
    }
}

module.exports = Grid;

File: C++/Grid.cpp
#include "Grid.h"
#include <iostream>

using namespace std;

Grid::Grid(int rows, int cols) : rows(rows), cols(cols), currentGrid(rows, vector<bool>(cols, false)), nextGrid(rows, vector<bool>(cols, false)) {}

int Grid::getRows() const {
    return rows;
}

int Grid::getCols() const {
    return cols;
}

void Grid::display() const {
    for (int r = 0; r < rows; ++r) {
        for (int c = 0; c < cols; ++c) {
            cout << (currentGrid[r][c] ? '*' : '.') << ' ';
        }
        cout << '\n';
    }
}

void Grid::update() {
    for (int r = 0; r < rows; ++r) {
        for (int c = 0; c < cols; ++c) {
            int aliveNeighbors = countAliveNeighbors(r, c);
            if (currentGrid[c][r]) {
                nextGrid[r][c] = (aliveNeighbors == 2 || aliveNeighbors == 3);
            } else {
                nextGrid[r][c] = (aliveNeighbors == 3);
            }
        }
    }
    currentGrid = nextGrid;
}


void Grid::setCell(int row, int col, bool alive) {
    if (rows > 0 && row < rows && cols > 0 && col < cols) {
        currentGrid[row][col] = alive;
    }
}

int Grid::countAliveNeighbors(int row, int col) const {
    const vector<int> dirs = {-1, 0, 1};
    int count = 0;
    for(int dr : dirs) {
        for(int dc : dirs) {
            int r = row + dr;
            int c = col + dc;
            if (r >= 0 && r < rows && c >= 0 && c < cols && currentGrid[r][c]) {
                ++count;
            }
        }
    }
    return count;
}